docs/design-docs/cli/cli-ux.md

```diff
@@ -0,0 +1,378 @@
+# Epiphany CLI UX
+
+Affected version: unknown
+
+## Goals
+
+This document aim is to improve user experience with epicli tool with strong emphasis to lower entry level for new users. It provides idea for following scenarios: 
+ * epicli installation
+ * environment initialization and deployment
+ * environment component update
+ * cli tool update
+ * add component to existing environment
+ 
+## Assumptions
+
+Following scenarios assume: 
+ * there is component version introduced - epicli version is separated from component version. It means that i.e. epicli v0.0.1 can provide component PostgreSQL 10.x and/or PostgreSQL 11.x.
+ * there is server-side component - epiphany environment is always equipped with server side daemon component exposing some API to epicli. 
+ 
+## Convention
+I used square brackets with dots inside: 
+```
+[...]
+```
+to indicate processing or some not important for this document output. 
+
+## Story
+
+### epicli installation
+
+To increase user base we need to provide brew formulae to allow simple installation. 
+```
+> brew install epicli
+``` 
```

Comment:
Good point about dependencies. However, this design does not consider technical implementation. I used "brew" to highlight importance of smooth user experience. It's super nice when all you need to do before you use tool is `brew install tool-you-want` (ex.: https://learn.hashicorp.com/terraform/getting-started/install.html#install-terraform). I'm okay with situation when we in fact run `docker run something` inside `epicli something`.

Reply:
I think we can say that dependencies are "implementation detail" (it can be tackled by using docker image underneath - just an idea to show there will be solutions for that).